### PR TITLE
GraalVM JS on Nextgen

### LINKED ISSUE #398: [UPGRADE] Upgrade JS Engine to GraalJS
We hope to upgrade our ScriptAPI for Nextgen to [GraalJS](https://github.com/oracle/graaljs), which is a high performance implementation of the JavaScript programming language. It's also compatible with the [ECMAScript 2020](https://262.ecma-international.org/11.0/) specification.
The main issue with using a new engine is the remapping of Minecraft. It took us a while to figure out how to remap nashorn.
Now we decided to try GraalJS.

### PR DESCRIPTION
**Pull request by 1zun4secondary**
We hope to upgrade our ScriptAPI for Nextgen to [GraalJS](https://github.com/oracle/graaljs), which is a high performance implementation of the JavaScript programming language. It's also compatible with the [ECMAScript 2022](https://262.ecma-international.org/13.0/) specification.
The main issue with using a new engine is the remapping of Minecraft. It took us a while to figure out how to remap nashorn.
Now we decided to try GraalJS.

### CODE REVIEW

**Comment by CzechHek** [comment]
😳

**Comment by 1zun4secondary** [comment]
grafik
grafik
grafik

**Comment by 1zun4secondary** [comment]
grafik

whoop whoop

**Comment by CzechHek** [comment]
> 😳

😳😳

**Comment by FaaatPotato** [comment]
damn 👾

**Comment by 1zun4secondary** [comment]
now only need to figure out why remapJar is taking so long that it times out when importing GraalJS library...

**Comment by 1zun4secondary** [comment]
grafik

Remapper works. Thanks a lot to the lit who contributed it.

**Comment by 1zun4secondary** [comment]
Commands are fully supported now as well!

grafik
grafik

**Comment by 1zun4** [comment]
done done

**Comment by 1zun4secondary** [comment]
It is now only missing support for module settings, which is quite annoying to implement and probably something for later. But besides that this implementation is done and can now be merged into the main branch.

File: src/main/kotlin/net/ccbluex/liquidbounce/script/bindings/features/JsSetting.kt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
  * This file is part of LiquidBounce (https://github.com/CCBlueX/LiquidBounce)
  *
- * Copyright (c) 2016 - 2022 CCBlueX
+ * Copyright (c) 2016 - 2021 CCBlueX
```
Comment (banocean) [inline src/main/kotlin/net/ccbluex/liquidbounce/script/bindings/features/JsSetting.kt:4]: 2021 not 2023?

Comment (1zun4secondary) [reply src/main/kotlin/net/ccbluex/liquidbounce/script/bindings/features/JsSetting.kt:4]: uhm oh?

**Pull request merged**